src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -87,6 +70,44 @@ def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[st
         }
         return config_file, sorted(deps)
 
+    def prepare_certificates(self) -> Tuple[str, str]:
+        cert = self.mgr.get_store('grafana_crt')
+        pkey = self.mgr.get_store('grafana_key')
+        (org, cn) = (None, None)
+        if cert and pkey:
+            try:
+                (org, cn) = get_cert_issuer_info(cert)
+                verify_tls(cert, pkey)
+            except ServerConfigException as e:
+                logger.warning(f'Provided grafana TLS certificates are invalid: {e}')
+                cert, pkey = None, None
```

Comment:
While I'm pretty sure you're just retaining old behavior here, I feel like it would be nice if we had some bool or something to mark that our cert/key failed verification. Right now, the condition `cert and pkey` failing can mean two different things: 1) There was no cert/key 2) there was a cert/key but it failed validation. Making it more explicit which of these two cases we're in I think would make it easier to verify the correctness of the cases in the following if/else block. Personally I find
```
if not (cert and pkey) \
           and ((org == 'Ceph' and cn == 'cephadm') or (org is None and cn is None)):
```
to be a bit of a confusing (although possibly entirely correct) check.

Reply:
I fixed the logic to be more explicit. Now two new bool variables are used to differentiate between the cases of: empty certificates and invalid certificates.

Reply:
New checks are much more clear.